src/saving.jl

```diff
@@ -433,7 +433,10 @@ function LinearizingSavingCallback(ils::IndependentlyLinearizedSolution{T,S};
         end,
         # We need to finalize the ils and free our caches
         finalize = (c, u, t, integ) -> begin
-            finish!(ils)
+            # Don't run the `finish!` if ils is in an inconsistent state
+            if check_error(integ) != ReturnCode.InitialFailure
```

Comment:
@staticfloat  is this the best thing to be checking or should I check something in `ils` or perhaps check if `t==0`?

Reply:
I think actually we still want to call finalize (so that various caches can be released, see the other PR in this repo adding even more caches) but we want to check this condition within the finish() function, and if we have errored, we want to return an empty ILS, rather than trying to collect all the u and t chunks.

Reply:
Cool. I will make that change now